### ISSUE TITLE
snowflake_stage recreates every apply regardless of an actual change

### ISSUE THREAD

**Issue by bmhah** [Snowflake-Labs/terraform-provider-snowflake#2679]
### Terraform CLI and Provider Versions

v1.5.0 
0.85.0

### Terraform Configuration

```terraform
resource "snowflake_stage" "LANDING_8X8_STAGE_INBOUND_CALLS_S3" {
  name                = "LANDING_8X8_STAGE_INBOUND_CALLS"
  url                 = var.s3_uri_8x8_inbound_calls
  database            = "AWS_LANDING_INGEST_DB_${var.SF_ENVIRONMENT}"
  schema              = "LANDING_8X8_SCHEMA"
  comment             = "Landing stage for inbound calls s3 bucket"
  file_format         = "FORMAT_NAME = AWS_LANDING_INGEST_DB_${var.SF_ENVIRONMENT}.LANDING_8X8_SCHEMA.LANDING_FF_CSV"
  copy_options        = "ON_ERROR = 'CONTINUE'"
  storage_integration = var.aws_storage_integration
}
```

### Expected Behavior

Stage doesn't destroy and recreate

### Actual Behavior

Stage recreates, this causes snowflake pipes to break having a status `STOPPED_STAGE_DROPPED`

### Steps to Reproduce

1. `terraform apply`

### How much impact is this issue causing?

High

**Comment by sfc-gh-asawicki** [comment]
Hey @bmhah. Thanks for reaching out to us.

Please share the terraform run logs with `TF_LOG=DEBUG` enabled.

**Comment by sfc-gh-bhill** [comment]
[plan.txt.gz](https://github.com/Snowflake-Labs/terraform-provider-snowflake/files/14984065/plan.txt.gz)

**Comment by sfc-gh-asawicki** [comment]
Hey @bmhah. I will confirm in the morning, but it looks like an easy fix, so we may even be able to include it in tomorrow's release. I will keep you posted.

**Comment by bmhah** [comment]
Sounds good, we'll get to testing this later this week. I'll keep this issue open in the meantime

**Comment by sfc-gh-asawicki** [comment]
Hey @bmhah. Have you have a chance to test the newer versions after the introduced fixes?

**Comment by bmhah** [comment]
Yes, we've done the provider upgrade to 88 and have also used others, they haven't worked for us, the stage for this pipe still drops and recreates causing the pipe to break again

**Comment by sfc-gh-asawicki** [comment]
The fix was introduced in 0.89, not 0.88, so please provide the config and logs (run terraform with TF_LOG=DEBUG) on the 0.89+ version (I could not reproduce the issue given your config on the 0.89 version).

**Comment by guillaumelecerf** [comment]
We were having the same issue with:
```
file_format         = "FIELD_DELIMITER = '|' PARSE_HEADER = true"
```

Fix in https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/2885

**Comment by sfc-gh-asawicki** [comment]
The fix was included in v0.93.0. Please check it @guillaumelecerf @bmhah.

**Comment by harukitake** [comment]
Hey @sfc-gh-asawicki.
I have the same issue on the version 0.93.0.

terraform configuration
```tf
resource "snowflake_stage" "stage" {
      database = "DATABASE_NAME"
      file_format = "TYPE = CSV NULL_IF = []"
      name = "NAME"
      schema = "SCHEMA_NAME"
      storage_integration = "INTEGRATION_NAME"
      url = "s3://aaaaa"
    }
```

**Comment by sfc-gh-asawicki** [comment]
I will take a look this week. Thanks for providing the config; it helps reproduce the issues.

**Comment by harukitake** [comment]
Hi @sfc-gh-asawicki.
Do you have any updates for this issue?

**Comment by sfc-gh-asawicki** [comment]
Hey @harukitake. Sorry, I missed this one, I will check it this week.

**Comment by sfc-gh-asawicki** [comment]
Hey @harukitake. I've finally reproduced the error.

The current implementation of the stage resource filters out all the file format properties that have default values in Snowflake. TYPE = CSV is the default value, so after reading the resource's state, it is programmatically removed from the file format string, which causes the permadiff you see.

This is, of course, not the expected behavior. For now, please just remove TYPE = CSV from your config, because this is the default in Snowflake.

We will fix the handling of these parameters during the stage resource rework in the next few weeks. I have added a test to confirm it and the note in the resource documentation.

**Comment by sfc-gh-asawicki** [comment]
Hey, this issue is part of the work planned for the next weeks. See more context in the [new roadmap entry](https://github.com/snowflakedb/terraform-provider-snowflake/blob/main/ROADMAP.md#2025-12-18-roadmap-update).

**Comment by sfc-gh-jmichalak** [comment]
Hi, we've released v2.13.0 ([release](https://github.com/snowflakedb/terraform-provider-snowflake/releases/tag/v2.13.0), [migration guide](https://github.com/snowflakedb/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v212x--v2130)) with new stage resources and a reworked data source. Enjoy!

**Comment by sfc-gh-jmichalak** [comment]
Closing due to inactivity.
